tools/common_lib/src/gemm.h

```diff
@@ -1327,6 +1327,49 @@ class GemmUmdD3d12Dispatcher : public GemmBaseDispatcher
         gemm_.execute(stream, args);
     }
 
+
+    ConformanceResult validate_conformance(ID3D12CommandQueue* command_queue,
+            ID3D12CommandAllocator* command_allocator, ID3D12GraphicsCommandList* command_list, bool print_mismatche) override
+    {
+        auto dump_buffer_to_file = [&](const auto& buffer, const auto& file_name)
+        {
+            if (!buffer)
+            {
+                return;
+            }
+            const auto bytes_width = buffer->GetDesc().Width;
+            // readback data and validate
+            auto readback_buffer = create_buffer(d3d12_device_, bytes_width, D3D12_HEAP_TYPE_READBACK, D3D12_RESOURCE_STATE_COPY_DEST);
+            auto readback_output_barrirer = CD3DX12_RESOURCE_BARRIER::Transition(buffer.Get(),
+                D3D12_RESOURCE_STATE_UNORDERED_ACCESS, D3D12_RESOURCE_STATE_COPY_SOURCE);
+            command_list->ResourceBarrier(1, &readback_output_barrirer);
+            command_list->CopyResource(readback_buffer.Get(), persistent_buffer_.Get());
+            close_execute_reset_wait(d3d12_device_, command_queue, command_allocator, command_list);
+
+            std::vector<std::byte> data_out(bytes_width);
+            std::byte* readback_mapped_ptr = nullptr;
+            readback_buffer->Map(0, nullptr, reinterpret_cast<void**>(&readback_mapped_ptr));
+            std::memcpy(data_out.data(), readback_mapped_ptr, data_out.size());
+            readback_buffer->Unmap(0, nullptr);
+
+             // Assuming data_out now contains the float data
+            float* float_ptr = reinterpret_cast<float*>(data_out.data());
```

Comment:
Float assumption is fine, but probably not always valid as scratchpad can have arbitrary data type <- you may want to add such extended comment so it will be clear for code reader in future.